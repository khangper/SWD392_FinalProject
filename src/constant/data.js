export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
