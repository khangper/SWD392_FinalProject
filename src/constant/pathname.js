export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
