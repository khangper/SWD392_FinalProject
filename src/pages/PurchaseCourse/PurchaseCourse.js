import React from "react";
import DeleteMyCourse from "..//../assets/garbage.png";
import EditMyCourse from "..//../assets/Edit-mycourse.png";
import MyCourseBook from "..//../assets/MyCourse-Book.png";
import DowLoadMyCourse from "..//../assets/DowLoadMyCourse.png";
import Printer from "..//../assets/Printer.png";
import { Link } from "react-router-dom";
import { PATH_NAME } from "../../constant/pathname";

function PurchaseCourse() {
  // My Purchase
  const MyPurchase = [
    {
      itemNo: "1",
      title: "Course Title Here",
      Vendor: "Zoena Singh",
      Category: "Web Development",
      DeliveryType: "Download",
      Price: "$15",
      Purchase: "25 March 2020",
      actionEdit: "#",
      actionDelete: "#",
    },
    {
      itemNo: "2",
      title: "Course Title Here",
      Vendor: "Rock William",
      Category: "C++",
      DeliveryType: "Download",
      Price: "$20",
      Purchase: "25 March 2020",
      actionEdit: "#",
      actionDelete: "#",
    },
  ];
  return (
    <div className="PurchaseCoursePage">
      <div className="MyCoursePage-grid-container">
        <div className="MyCoursePage-grid-item">
          <div className="MyCoursePage-st_title">
            <img src={MyCourseBook} className="Edit-icon" />
            Purchased Course
          </div>
        </div>
      </div>

      {/* Body */}

      <div className="MyCouser-containerr">
        <div className="MyCouser-table-container">
          <table className="MyCourse-table ucp-table">
            <thead className="MyCourse-thead-s">
              <tr>
                <th className="MyCourse-text-center" scope="col">
                  Item No.
                </th>
                <th className="MyCourse-text-center">Title</th>
                <th className="MyCourse-text-center" scope="col">
                  Vendor
                </th>
                <th className="MyCourse-text-center" scope="col">
                  Category
                </th>
                <th className="MyCourse-text-center" scope="col">
                  Delivery Type
                </th>
                <th className="MyCourse-text-center" scope="col">
                  Price
                </th>
                <th className="MyCourse-text-center" scope="col">
                  Purchase Date
                </th>
                <th className="MyCourse-text-center" scope="col">
                  Action
                </th>
              </tr>
            </thead>
            <tbody>
              {MyPurchase.map((course) => (
                <tr key={course.itemNo}>
                  <td className="MyCourse-text-center">{course.itemNo}</td>
                  <td>{course.title}</td>
                  <td className="MyCourse-text-center">
                    <a href="#">{course.Vendor} </a>
                  </td>
                  <td className="MyCourse-text-center">
                    <a href="#">{course.Category} </a>
                  </td>
                  <td className="MyCourse-text-center">
                    <Link to={PATH_NAME.DOWLOADCOURSE}>
                      <b className="MyCourse-course_active">
                        {course.DeliveryType}
                      </b>
                    </Link>
                  </td>
                  <td className="MyCourse-text-center">{course.Price}</td>
                  <td className="MyCourse-text-center">{course.Purchase}</td>
                  <td className="MyCourse-text-center">
                    <img src={DowLoadMyCourse} className="Edit-icon" />
                    <img src={DeleteMyCourse} className="Edit-icon" />
                    <img src={Printer} className="Edit-icon" />
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
}

export default PurchaseCourse;
