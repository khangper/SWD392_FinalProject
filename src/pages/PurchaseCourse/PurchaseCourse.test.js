import React from "react";
import { render, screen } from "@testing-library/react";
import PurchaseCourse from "../PurchaseCourse/PurchaseCourse";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("PurchaseCourse", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the PurchaseCourse component", async () => {
    const { container } = renderWithProviders(<PurchaseCourse />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
