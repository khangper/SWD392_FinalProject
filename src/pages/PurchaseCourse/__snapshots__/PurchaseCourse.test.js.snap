// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseCourse renders the PurchaseCourse component 1`] = `
<div>
  <div
    class="PurchaseCoursePage"
  >
    <div
      class="MyCoursePage-grid-container"
    >
      <div
        class="MyCoursePage-grid-item"
      >
        <div
          class="MyCoursePage-st_title"
        >
          <img
            class="Edit-icon"
            src="MyCourse-Book.png"
          />
          Purchased Course
        </div>
      </div>
    </div>
    <div
      class="MyCourse-container"
    >
      <div
        class="MyCourse-table-container"
      >
        <table
          class="MyCourse-table ucp-table"
        >
          <thead
            class="MyCourse-thead-s"
          >
            <tr>
              <th
                class="MyCourse-text-center"
                scope="col"
              >
                Item No.
              </th>
              <th
                class="MyCourse-text-center"
              >
                Title
              </th>
              <th
                class="MyCourse-text-center"
                scope="col"
              >
                Vendor
              </th>
              <th
                class="MyCourse-text-center"
                scope="col"
              >
                Category
              </th>
              <th
                class="MyCourse-text-center"
                scope="col"
              >
                Delivery Type
              </th>
              <th
                class="MyCourse-text-center"
                scope="col"
              >
                Price
              </th>
              <th
                class="MyCourse-text-center"
                scope="col"
              >
                Purchase Date
              </th>
              <th
                class="MyCourse-text-center"
                scope="col"
              >
                Action
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
